.github/workflows/build_steps.yml

```diff
@@ -299,6 +337,9 @@ jobs:
           # Use the Mongo created in the service container above to test against
           CI_MONGO_HOST: mongodb
           CI_MONGO_HOST_2: mongodb_2
+          # On windows, we can allocate more threads for running the tests
+          # because we are not testing against mongo
```

Comment:
This is an incorrect assumption/misleading comment.

We did not restrict the tests to two processes because of Mongo. Rather, the current mongo fixture only supports two processes because the default free runner only has two cores.

The four-core machine we sometimes get is a random free upgrade from GitHub and we can't rely on always getting that.

Reply:
In my testing, **running test mongo tests in parallel makes them break consistently**.
That is what my comment is referring to and to the fact that we cannot truly run the tests in parallel before this issue is fixed.
I have opened an issue to fix this limitation.

Reply:
>In my testing, running test mongo tests in parallel makes them break consistently.

Oh, that might be because you're not running it with `parallel-test.sh`?

The current Mongo fixture relies on [parsing a pytest-split-specific argument](https://github.com/man-group/ArcticDB/blob/master/python/tests/conftest.py#L562) to select one of the two Mongo servers to use.

My new fixture will replace this dodgy hack